### PR TITLE
[Bug] Can not use non-key column as partition column in duplicate table

### PR DESCRIPTION
**Pull request by morningman**
The following statement will throw error:
```
create table test.tbl2
(k1 int, k2 int, k3 float)
duplicate key(k1)
partition by range(k2)
(partition p1 values less than("10"))
distributed by hash(k3) buckets 1
properties('replication_num' = '1'); 
```
Error: `Only key column can be partition column`

But in duplicate key table, columns can be partition or distribution column
even if they are not in duplicate keys.

This bug is introduced by #3812

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/analysis/RangePartitionDesc.java [hunk 1]
```diff
@@ -145,9 +149,14 @@ public PartitionInfo toPartitionInfo(List<Column> schema, Map<String, Long> part
             boolean find = false;
             for (Column column : schema) {
                 if (column.getName().equalsIgnoreCase(colName)) {
-                    if (!column.isKey()) {
+                    if (!column.isKey() && column.getAggregationType() != AggregateType.NONE) {
```
Comment (EmmyMiao87) [inline fe/src/main/java/org/apache/doris/analysis/RangePartitionDesc.java:152]: Why does it need a second check here?

Comment (morningman) [reply fe/src/main/java/org/apache/doris/analysis/RangePartitionDesc.java:152]: Historical issues...
It is OK, may be removed in future.

File: fe/src/main/java/org/apache/doris/analysis/RangePartitionDesc.java [hunk 2]
```diff
@@ -145,9 +149,14 @@ public PartitionInfo toPartitionInfo(List<Column> schema, Map<String, Long> part
...
                         throw new DdlException("Partition column[" + colName + "] is not key column");
```
Comment (EmmyMiao87) [inline fe/src/main/java/org/apache/doris/analysis/RangePartitionDesc.java:153]:
```suggestion
                        throw new DdlException("The partition column could not be aggregated column");
```

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**